platform/darwin/scripts/generate-style-code.js

```diff
@@ -4,11 +4,26 @@ const fs = require('fs');
 const ejs = require('ejs');
 const _ = require('lodash');
 const colorParser = require('csscolorparser');
-const spec = _.merge(require('mapbox-gl-style-spec').latest, require('./style-spec-overrides-v8.json'));
-
+const cocoaConventions = require('./style-spec-cocoa-conventions-v8.json');
+let spec = _.merge(require('mapbox-gl-style-spec').latest, require('./style-spec-overrides-v8.json'));
 const prefix = 'MGL';
 const suffix = 'StyleLayer';
 
+function renameProperties(obj, overrides, stack='') {
+    _.forOwn(overrides, function (value, key) {
+        const keyPath = stack + '.';
+        if (_.isObject(value)) {
+            renameProperties(obj, overrides[key], stack.length ? keyPath + key : key);
```

Comment:
Does this function need to be recursive? Seems like we could assume a two-level-deep override object and use subscript notation instead of _.set and _.unset.